File: 23-09-26/4.py
#code by pranay
#date 23-09-26

#defining the required  function, that checks the stable signal and returns the inddex
def firstStable(a, n, tolerance):
    for i in range(n - 1):
        #checking if the difference is less than the rolerable number
        if abs(a[i + 1] - a[i]) <= tolerance:
            return i
        #return the negative stability as -1
    return -1


#reading the number string
n = int(input())
#converting number string into array
a = list(map(float, input().split()))
tolerance = float(input())

print(firstStable(a, n, tolerance))

File: 23-09-26/4.c
//code by pranay
//date 23-09-26

#include <stdio.h>
#include <math.h>

//checks the first stable point in the array and returns the index of the first stable point
int firstStable(double a[], int n, double tolerance)
{
    for (int i = 0; i < n - 1; i++)
    {//Checks if the diff is less than the tolerance value
        if (fabs(a[i + 1] - a[i]) <= tolerance)
            return i;
    }
	//else return no stable (-1 for my reference)
    return -1;
}

int main()
{
    int n;
    double tolerance;
	printf("Enter the number of readings: ");
    scanf("%d", &n);

    double a[n];

    printf("Input the reading array: ");
    for (int i = 0; i < n; i++)
        scanf(" %lf", &a[i]);

    printf("Enter the tolerance for difference: ");
    scanf("%lf", &tolerance);

    printf("%d\n", firstStable(a, n, tolerance));

    return 0;
}
